src/meta-srv/src/region/lease_keeper.rs

```diff
@@ -59,33 +48,103 @@ impl RegionLeaseKeeper {
             table.push(*region_id);
         }
 
-        let table_ids = tables.keys().cloned().collect::<Vec<_>>();
+        tables
+    }
+
+    async fn collect_table_metadata(
+        &self,
+        table_ids: &[TableId],
+    ) -> Result<HashMap<TableId, TableRouteValue>> {
+        let table_route_manager = self.table_metadata_manager.table_route_manager();
 
         // The subset of all table metadata.
         // TODO: considers storing all active regions in meta's memory.
         let metadata_subset = table_route_manager
-            .batch_get(&table_ids)
+            .batch_get(table_ids)
```

Comment:
How many tables' metadata will be loaded at once?

Reply:
Typically, the number is the all leader(follower) regions at once. We might need to maintain a small piece of metadata in the leader's memory to track all expected/opening regions of each datanodes. No more worry it doesn't hit the cache.